pkg/controller/template/render.go

```diff
@@ -379,7 +379,7 @@ func cloudConfigFlag(cfg RenderConfig) interface{} {
 		}
 	}
 
-	external, err := cloudprovider.IsCloudProviderExternal(cfg.Infra.Status.PlatformStatus.Type, cfg.FeatureGate)
+	external, err := cloudprovider.IsCloudProviderExternal(cfg.Infra.Status.PlatformStatus, cfg.FeatureGate)
```

Comment:
For my own info, can you explain this change?

Reply:
We had to change the signature of the upstream method to include more information from the Infrastructure resource to allow the method to determine whether it is on a special cloud, in this case Azure stack hub (this also applies for things like gov clouds etc), where the platform shows up as Azure, but the designation that this is Azure Stack Hub is only available by looking at the Azure specific fields on the `PlatformStatus`. We don't expect this to change again in the future as we now have access to all of the provider specific statuses to make the decision needed in the future

Reply:
thanks for the details @JoelSpeed makes sense to me.

Reply:
as yes it's right here :smile: 
https://github.com/openshift/library-go/pull/1077/files#diff-478af36e9fb994fc80d37b7d2f6ae207c67d8c43b94f98f6ae3e420808958ba9